tests/rptest/scale_tests/rpk_many_partitions_test.py

```diff
@@ -0,0 +1,144 @@
+
+            # Consume messages from the topic with
+            # `rpk topic consume`.
+            rpk_consumer = RpkConsumer(self.test_context,
+                                       self.redpanda,
+                                       save_msgs=False,
+                                       retry_sec=0.5,
+                                       topic=topic_spec.name,
+                                       group=group_name)
+            rpk_consumer.start()
+
+            wait_until(lambda: rpk_consumer.message_count >= msg_count,
+                       timeout_sec=120,
+                       backoff_sec=5,
+                       err_msg="RPK failed to fetch all messages")
+
+            self.logger.debug(
+                f"RPK successfully fetched {rpk_consumer.messages}")
+
+            # `rpk topic consume` is in a loop that will
+            # run forever unless we call stop here
+            rpk_consumer.stop()
+
+            # `rpk topic produce` should be done already but
+            # we call wait and stop just to be sure
+            rpk_producer.wait()
+            rpk_producer.stop()
+
+        def _describe_topic(topic_name, partition_count):
+            try:
+                partitions = list(
+                    self._rpk.describe_topic(topic_name, tolerant=True))
+            except RpkException as e:
+                # Retry once.
+                self.logger.error(f"Retrying describe_topic for {e}")
+                partitions = list(
+                    self._rpk.describe_topic(topic_name, tolerant=True))
+
+            if len(partitions) < partition_count:
+                self.logger.info(
+                    f"rpk topic describe omits partitions for topic {topic_name}"
+                )
+
+            return len(partitions) == partition_count
+
+        # Producing 1K messages to a 50K partitions topic.
+        _produce_and_consume()
+
+        # Describe the topic with `rpk topic describe`
+        wait_until(
+            lambda: _describe_topic(topic_spec.name, partition_count_1),
+            timeout_sec=60,
+            backoff_sec=1,
+            err_msg=
+            f"RPK failed to describe topic {topic_spec.name} with {partition_count_1} partitions"
+        )
+
+        # Then we add the rest of the partitions to get to 80K.
+        self._rpk.add_partitions(topic_spec.name,
+                                 partition_count_2 - partition_count_1)
+
+        # Producing another 1K messages but with 80K partitions.
+        _produce_and_consume()
+
+        wait_until(
+            lambda: _describe_topic(topic_spec.name, partition_count_2),
+            timeout_sec=60,
+            backoff_sec=1,
+            err_msg=
+            f"RPK failed to describe topic {topic_spec.name} with {partition_count_2} partitions"
+        )
+
+        # Check the output of `rpk cluster health`.
+        health = self._rpk.cluster_health()
+        assert len(health.nodes_down) == 0
```

Comment:
Add a message so we know where the assert happened if this line ever fails
```suggestion
        assert len(health.nodes_down) == 0, "rpk cluster health failed: too many downed nodes"
```

Reply:
Also, you are checking for `nodes_down` here. Are there any other values that we should also check here?

Reply:
In the past, I had a check for the is_healthy value, but we actually don't care if the cluster is unhealthy (_for rpk test only_), we just want to check how rpk behaves in a cluster of this size.